src/main.rs

```diff
@@ -161,6 +167,23 @@ fn check_host(target: &CanaryTarget) -> CanaryCheck {
     }
 }
 
+
+/*
+    Checks if alert would not be spam.
+    Alert would not be spam iff state changes from "okay" to "fire" or "fire" to "okay".
+    Updates HashMap last_statuses with last state seen for the target.
+*/
+fn check_not_spam(last_statuses: &mut HashMap<CanaryTarget, String>, result: &CanaryCheck) -> bool {
+    let not_spam = match last_statuses.get(&result.target) {
+        Some(status) => {
+            (status == "okay" && result.info == "fire") || (status == "fire" && result.info == "okay")
```

Comment:
`unknown` status is also not spam. I will change it to use a status enum in the future, but a check for an `unknown` string is needed here for now.

An easy way to test this is to run a local server with `python -m SimpleHTTPServer 8000`, and changing the config file to `host = "http://0.0.0.0:8000"`.